### PR TITLE
Maven task does not find the task-report.txt is some cases

### PR DESCRIPTION
**Pull request by bgavrilMS**
https://github.com/Microsoft/vsts-tasks/issues/2485

### CODE REVIEW

File: Tasks/Gradle/CodeAnalysis/SonarQube/common.ts [hunk 1]
```diff
@@ -40,9 +41,15 @@ export function applySonarQubeParameters(toolRunner: ToolRunner): ToolRunner {
  * @param sqBuildFolder
  * @returns {Promise<void>} Promise resolved when all SQ integration actions are complete.
  */
-export function processSonarQubeIntegration(sqBuildFolder:string):Q.Promise<void> {
-    var sqRunSettings:SonarQubeRunSettings = getSonarQubeRunSettings(sqBuildFolder);
-    var sqMetrics:SonarQubeMetrics = getSonarQubeMetrics(sqRunSettings);
+export function processSonarQubeIntegration(sqTaskReportGlob: string): Q.Promise<void> {
```
Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/SonarQube/common.ts]: The main entry point into the post processing we do around SQ changes to accept a glob path. Note that is code is common to both Gradle and Maven and it is build agnostic.

File: Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -80,6 +80,8 @@
   "loc.messages.sqAnalysis_QualityGatePassed": "The SonarQube quality gate associated with this build has passed (status %s)",
   "loc.messages.sqAnalysis_UnknownComparatorString": "The SonarQube build summary encountered a problem: unknown comparator '%s'",
   "loc.messages.sqAnalysis_NoUnitsFound": "The list of SonarQube measurement units could not be retrieved from the server.",
+  "loc.messages.sqAnalysis_NoReportTask": "Could not find report-task.txt. Possible cause: the SonarQube analysis did not complete successfully.",
+  "loc.messages.sqAnalysis_MultipleReportTasks": "Multiple report-task.txt files found. Choosing the first one. The build summary and the build breaker may not be accurate. Possible cause: multiple SonarQube analysis during the same build, which is not supported.",
```
Comment (HSAR) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: I'm not aware that this limitation is noted anywhere in the docs, do they say anything about this?

File: Tasks/Gradle/CodeAnalysis/SonarQube/run-settings.ts [hunk 3]
```diff
@@ -50,7 +50,8 @@ export class SonarQubeRunSettings {
      * @returns {SonarQubeRunSettings} A new SonarQubeRunSettings with appropriate fields filled
      */
     public static createRunSettingsFromFile(filePath:string): SonarQubeRunSettings {
-        if (!tl.exist(filePath)) {
+        
+        if (!fs.existsSync(filePath)) {
```
Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/SonarQube/run-settings.ts]: This function is deprecated. See the following:

https://nodejs.org/api/fs.html#fs_fs_existssync_path
http://stackoverflow.com/questions/28532820/fs-exists-fs-existssync-why-are-they-deprecated

File: Tasks/Gradle/CodeAnalysis/SonarQube/common.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 import Q = require('q');
 import path = require('path');
+import glob = require('glob');
```
Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/SonarQube/common.ts]: Yes. Glob is used extensively throught the tasks.

File: Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson (hunk 2 above)
Comment (bgavrilMS) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: It is self documented through the warming message. The docs are general purpose and do not list obscure scenarios.

File: Tasks/Gradle/CodeAnalysis/SonarQube/run-settings.ts (hunk 3 above)
Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/SonarQube/run-settings.ts]: Looking at this.

File: Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson (hunk 2 above)
Comment (HSAR) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: If a user wished to try this, the first time they are going to know that it isn't supported is when their build fails. This doesn't sound like a good experience to me.

Comment (bgavrilMS) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: It isn't going to be a good experience for them anyway - their test results will be all mixed up and their code coverage data confused. The point is to give the user enough information to log an issue.

Comment (HSAR) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: That's why I'm asking if the online docs say anything, so they don't have to get to that point before they find out it's not supported.

Comment (bgavrilMS) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: I'm pretty sure I already answered your question.

File: Tasks/Gradle/CodeAnalysis/SonarQube/run-settings.ts [hunk 5]
```diff
@@ -49,8 +49,10 @@ export class SonarQubeRunSettings {
      * @param filePath Path to a report-task.txt file generated by a SonarQube build plugin
      * @returns {SonarQubeRunSettings} A new SonarQubeRunSettings with appropriate fields filled
      */
-    public static createRunSettingsFromFile(filePath:string): SonarQubeRunSettings {
-        if (!tl.exist(filePath)) {
+    public static createRunSettingsFromFile(filePath: string): SonarQubeRunSettings {
+
+        let stats = fs.statSync(filePath);
```
Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/SonarQube/run-settings.ts]: Unused variable?

**Comment by arunjuly27** [comment]
Hi Team,

Looks like "Maven task does not find the task-report.txt is some cases" issue has been fixed.

Just wanted to know how to get the updated version in our TFS instance, we are using in on premise 

Thanks

**Pull request merged**